arcgishub/sites.py

```diff
@@ -413,11 +436,96 @@ def update(self, site_properties=None, data=None, thumbnail=None, metadata=None)
         
             >> True
         """
+        _site_data = self.definition
         if site_properties:
-            _site_data = self.definition
             for key, value in site_properties.items():
                 _site_data[key] = value
-            return self.item.update(_site_data, data, thumbnail, metadata)
+        if subdomain:
+            #format subdomain if needed
+            subdomain = subdomain.replace(' ', '-').lower()
+            #Domain manipulation for new subdomain
+            if self._gis._portal.is_arcgisonline:
+                #Check for length of domain
+                if len(subdomain + '-' + self._gis.properties['urlKey']) > 63:
+                    _num = 63 - len(self._gis.properties['urlKey'])
+                    raise ValueError('Requested url too long. Please enter a subdomain shorter than %d characters' %_num)
+                #Fetch siteId from domain entry
+                _HEADERS = {
+                    'Content-Type': 'application/json', 
+                    'Authorization': self._gis._con.token, 
+                    'Referer': self._gis._con._referer
+                }
+                path = 'https://hub.arcgis.com/utilities/domains?siteId='+self.itemid
+                _site_domain = requests.get(path, headers = _HEADERS)
+                _siteId = _site_domain.json()[0]['id']
+                client_key = _site_domain.json()[0]['clientKey']
+                #Delete old domain entry
+                _HEADERS = {
+                    'Content-Type': 'application/json', 
+                    'Authorization': self._gis._con.token, 
+                    'Referer': self._gis._con._referer
+                }
+                path = 'https://hub.arcgis.com/utilities/domains/'+_siteId
```

Comment:
Same as above. General advice: whenever there is a hard-coded string consider how to extract to a re-used variable to prevent bugs (mis-spelled strings) or changes (QA/Dev environments)